spicy/toolchain/src/compiler/printer.cc

```diff
@@ -16,6 +16,26 @@
 
 using namespace spicy;
 
+bool spicy::detail::printer::printID(hilti::printer::Stream& out, const ID& id) {
+    // In user-visible output, replace any `hilti` prefix with `spicy`. This is
+    // a bit of a hammer: it's assuming that any HILTI types showing up there
+    // have a corresponding Spicy type. The alternative would be to explicitly
+    // identify valid mappings somehow (e.g., through a shared `&cxxname`).
+    // However, that's neither easy nor is it clear that that's worth it. For
+    // one, we currently do indeed maintain only such 1:1 mappings (i.e., we
+    // don't rename IDs existing at both layers other than the namespace). And
+    // second, when displaying Spicy code to users, there should really never
+    // be any HILTI identifier showing up anyways; so if we still end up with
+    // any, printing them with a `spicy` prefix is probably still a better
+    // solution than just printing them as-is.
+    if ( out.state().user_visible && id.namespace_() && id.sub(0) == ID("hilti") ) {
```

Comment:
Entering nit territory:

```suggestion
    if ( out.state().user_visible && id.length() > 1 && id.sub(0) == ID("hilti") ) {
```

The semantics of `namespace_()` are always slightly surprising to me since it returns everything up to the local part, e.g., in `spicy::Charset::UTF8` the "namespace" is `spicy::Charset`. What we actually want here IDs with `length > 1`.

This would also avoid allocating a `string` for `namespace` when all we want to check is presence.

Reply:
I read the original versions "does it have a namespace?", which sounds right, doesn't matter how long it is, that's what the next check is for.